### PR TITLE
(#28) feat: response default page when page not found

### PR DESCRIPTION
**Pull request by shana0440**

### CODE REVIEW

File: router.go [hunk 1]
```diff
@@ -30,10 +31,13 @@ func (r *Route) Call(req *http.Request) interface{} {
 	}
 
 	handler := r.matching(rs)
+	if handler == nil {
+		return nil, errors.New("page not found")
```
Comment (dannypsnl) [inline router.go]: `errors.New("page not found")` this is not good enough for me, I will say create error constants is better.

Comment (shana0440) [reply router.go]: you're right.

File: router.go [hunk 2]
```diff
@@ -19,7 +20,7 @@ func NewRoute() *Route {
 	}
 }
 
-func (r *Route) Call(req *http.Request) interface{} {
+func (r *Route) Call(req *http.Request) (interface{}, error) {
```
Comment (dannypsnl) [inline router.go]: I think to refactor `Call` to `Handle`, send `http.ResponseWriter` into Handle, stop return `interface{}`
What do you say?

File: router.go [hunk 3]
```diff
@@ -32,7 +31,7 @@ func (r *Route) Call(req *http.Request) (interface{}, error) {
 
 	handler := r.matching(rs)
 	if handler == nil {
-		return nil, errors.New("page not found")
+		return nil, PageNotFound("can't found " + req.URL.Path)
```
Comment (dannypsnl) [inline router.go]: https://stackoverflow.com/questions/1760757/how-to-efficiently-concatenate-strings-in-go
nice to use more efficient way to concatenate the string

File: router.go (hunk 2 above)
Comment (shana0440) [reply router.go]: let router handle response is little weird for me.

File: router.go (hunk 3 above)
Comment (shana0440) [reply router.go]: I will create a function at helper to handle string concat.

File: router.go (hunk 2 above)
Comment (dannypsnl) [reply router.go]: Ok, that's fine. Seems NotFound is handling by Rocket, it would be strange exactly.

**Comment by dannypsnl** [comment]
resolve #28

**Pull request merged**